typed/clj.checker/src/typed/cljc/checker/free_ops.clj

```diff
@@ -67,37 +67,40 @@
 ;
 ;frees-map :- '{t/Sym '{:F F :bnds (U Bounds Regex TypeFn)}}
 (defn with-free-mappings
-  [opts frees-map]
-  (let [_ (assert (frees-map? frees-map) frees-map)
-        scoped-names (keys frees-map)
-        vals (vals frees-map)
-        fresh-names (mapv (comp :name :F) vals)
-        bndss (mapv :bnds vals)]
-    (-> opts
-        (tvar/with-extended-new-tvars scoped-names fresh-names)
-        (bnds/with-extended-bnds fresh-names bndss))))
+  ([opts frees-map]
+   (let [_ (assert (frees-map? frees-map) frees-map)
+         scoped-names (keys frees-map)
+         vals (vals frees-map)
+         fresh-names (mapv (comp :name :F) vals)
+         bndss (mapv :bnds vals)]
+     (-> opts
+         (tvar/with-extended-new-tvars scoped-names fresh-names)
+         (bnds/with-extended-bnds fresh-names bndss))))
+  ([opts fresh-names bndss]
+   (-> opts
+       (tvar/with-extended-new-tvars fresh-names)
```

Comment:
Here we're scoping fresh names instead of scoped names. Is this because they are the same?

Reply:
I suppose they are. The original 2-arity version of `with-free-mappings` usually received `frees-map` from `with-bounded-frees`. The keys in the `frees-map` were taken out of `F`s which in turn were constructed at the callsites with `make-F`. So, apparently, `scoped-names` and `fresh-names` are usually the same thing.

Reply:
There were a couple of calls that needed scoped names that I added failing tests for. Otherwise I think this is true for 90% of the occurrences in the code base.